lib/Doctrine/DBAL/Platforms/AbstractPlatform.php

```diff
@@ -787,7 +787,7 @@ public function getModExpression($expression1, $expression2)
      * Returns the SQL snippet to trim a string.
      *
      * @param string      $str  The expression to apply the trim to.
-     * @param int         $mode The position of the trim (leading/trailing/both).
+     * @param int|null    $mode The position of the trim (leading/trailing/both).
```

Comment:
Why null? It doesn't seem to have any special meaning here.

Reply:
I thought having to _specify_ `TrimMode::UNSPECIFIED` would be too ironic. The purpose is to specify a 3rd argument.

```
ERROR: NullArgument - tests/Doctrine/Tests/DBAL/Platforms/SQLAnywherePlatformTest.php:611:58 - Argument 2 of Doctrine\DBAL\Platforms\SQLAnywherePlatform::getTrimExpression cannot be null, null value provided to parameter with type int (see https://psalm.dev/057)
            $this->platform->getTrimExpression('column', null, 'c')
```

Reply:
The test is obviously wrong. The method doesn't accept NULL. There's no way in PHP to omit the 2nd optional argument and specify the 3rd one.

Reply:
Ok I'll use `TrimMode::UNSPECIFIED`, I just noticed it is used in the rest of the method.